kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/avro/AvroFormat.java

```diff
@@ -17,20 +17,30 @@
 package io.confluent.connect.s3.format.avro;
 
 import io.confluent.connect.avro.AvroData;
+import io.confluent.connect.avro.AvroDataConfig;
 import io.confluent.connect.s3.S3SinkConnectorConfig;
 import io.confluent.connect.s3.storage.S3Storage;
 import io.confluent.connect.storage.format.Format;
 import io.confluent.connect.storage.format.RecordWriterProvider;
 import io.confluent.connect.storage.format.SchemaFileReader;
 import io.confluent.connect.storage.hive.HiveFactory;
 
+import java.util.Map;
+
+import static io.confluent.connect.avro.AvroDataConfig.SCHEMAS_CACHE_SIZE_CONFIG;
+
 public class AvroFormat implements Format<S3SinkConnectorConfig, String> {
   private final S3Storage storage;
   private final AvroData avroData;
 
   public AvroFormat(S3Storage storage) {
     this.storage = storage;
-    this.avroData = new AvroData(storage.conf().getInt(S3SinkConnectorConfig.SCHEMA_CACHE_SIZE_CONFIG));
+    Map<String, Object> avroDataProps = io.confluent.connect.avro.AvroDataConfig.baseConfigDef()
```

Comment:
I feel that this code block here (parsing and overwriting the `AvroDataConfig` values) rather belongs in an `avroDataConfig` method within `S3SinkConnectorConfig` and not here.